### PR TITLE
Following-up improvments for various PRs

### PR DESCRIPTION
**Pull request by wxiaoguang**
For:

* #22743
* #25408
* #25412
* #25588

### CODE REVIEW

File: custom/conf/app.example.ini [hunk 1]
```diff
@@ -193,8 +193,8 @@ RUN_USER = ; git
 ;; Use `ssh-keygen` to parse public SSH keys. The value is passed to the shell. By default, Gitea does the parsing itself.
 ;SSH_KEYGEN_PATH =
 ;;
-;; Enable SSH Authorized Key Backup when rewriting all keys, default is true
-;SSH_AUTHORIZED_KEYS_BACKUP = true
+;; Enable SSH Authorized Key Backup when rewriting all keys, default is false
+;SSH_AUTHORIZED_KEYS_BACKUP = false
```
Comment (delvh) [inline custom/conf/app.example.ini:197]: Isn't that breaking?
Or did it previously display the wrong value?

File: models/git/branch.go [hunk 2]
```diff
@@ -101,9 +101,9 @@ func (err ErrBranchesEqual) Unwrap() error {
 type Branch struct {
 	ID            int64
 	RepoID        int64  `xorm:"UNIQUE(s)"`
-	Name          string `xorm:"UNIQUE(s) NOT NULL"`
+	Name          string `xorm:"UNIQUE(s) NOT NULL"` // git's ref-name is case-sensitive internally, however, in database, it's case-insensitive at the moment
 	CommitID      string
-	CommitMessage string `xorm:"TEXT"`
+	CommitMessage string `xorm:"TEXT"` // it only stores the message summary (the first line)
```
Comment (delvh) [inline models/git/branch.go]: :fearful:
Those two comments don't sound like a good idea…

File: custom/conf/app.example.ini (hunk 1 above)
Comment (wxiaoguang) [reply custom/conf/app.example.ini:197]: Set SSH_AUTHORIZED_KEYS_BACKUP to false #25412

File: models/git/branch.go (hunk 2 above)
Comment (wxiaoguang) [reply models/git/branch.go]: I don't like them, but that's the fact.

 Sync branches into databases #22743 
 Fix branch commit message too long problem #25588

File: custom/conf/app.example.ini (hunk 1 above)
Comment (lunny) [reply custom/conf/app.example.ini:197]: > Isn't that breaking? Or did it previously display the wrong value?

This is just a missed documentation from #25412

File: models/git/branch.go [hunk 3]
```diff
@@ -101,9 +101,9 @@ func (err ErrBranchesEqual) Unwrap() error {
 type Branch struct {
 	ID            int64
 	RepoID        int64  `xorm:"UNIQUE(s)"`
-	Name          string `xorm:"UNIQUE(s) NOT NULL"`
+	Name          string `xorm:"UNIQUE(s) NOT NULL"` // git's ref-name is case-sensitive internally, however, in database, it's case-insensitive at the moment
```
Comment (lunny) [inline models/git/branch.go]: It's not true. For mysql/mssql the default collations are case-insensitive, for sqlite/postgres/tidb, the default collations are case-sensitive. I will send a PR to fix that.

Comment (wxiaoguang) [reply models/git/branch.go]: yup, "database related": mssql, mysql , by default.

Comment (wxiaoguang) [reply models/git/branch.go]:
```suggestion
	Name          string `xorm:"UNIQUE(s) NOT NULL"` // git's ref-name is case-sensitive internally, however, in some databases (mssql, mysql, by default), it's case-insensitive at the moment
```

**Review by lunny — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]
🙏 thanks

**Pull request merged**